modules/imgproc/src/color_lab.cpp

```diff
@@ -978,8 +978,9 @@ static struct LUVLUT_T {
     const long long int *LvToVpl_b;
 } LUVLUT = {0, 0, 0};
 
+// Clamp a float value to [0,1]. If the value is NaN, arbitrarily choose 1.f.
 #define clip(value) \
-    value < 0.0f ? 0.0f : value > 1.0f ? 1.0f : value;
+    value < 0.0f ? 0.0f : value > 1.0f ? 1.0f : cvIsNaN(value) ? 1.f : value;
```

Comment:
all NaN comparison returns false, we can use that:

```
value < 0.0f ? 0.0f : value <= 1.0f ? value : 1.0f;
```

https://stackoverflow.com/questions/1565164/what-is-the-rationale-for-all-comparisons-returning-false-for-ieee754-nan-values

---

> If the value is NaN, arbitrarily choose 1.f.

Why is 1.0?
Does it work with 0.0?

Reply:
It works for anything within the allowed range [0,1].
We need to put something in the range and not a NaN so that it does not read out of memory later. Now, we can choose to return a bogus value for a NaN (this is the behavior that I chose because it requires the least change; I chose to be based on 1.f, but it could be anything in [0,1]), or an empty matrix, or even trigger a CV_Error.
It is a design decision for you to please take :)
Checking for NaNs in the first place would be a small overhead compared to the complexity of color conversions.

Reply:
> design decision

Declare NaNs as "undefined" behavior with undefined result. Prevent crashes with minimal performance changes.

I prefer this code: `value < 0.0f ? 0.0f : value <= 1.0f ? value : 0.0f;` with 0.0f result for out of range/NaN values.